client/src/common/components/Activities/components/ActivitiesList.jsx

```diff
@@ -19,47 +25,75 @@ class ActivitiesList extends PureComponent {
   };
 
   componentDidMount() {
-    const { fetchActivities } = this.props;
+    this.handleShowActivities();
+  }
 
-    this.setState({ pending: true });
+  componentWillUnmount() {
+    const { removeActivities } = this.props;
+
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
 
-    this.pendingPromise = makeAbortablePromise(fetchActivities());
+    removeActivities();
+  }
+
+  handleShowActivities = () => {
+    const { fetchActivities, nextPage } = this.props;
+
+    this.setState({ pending: true });
+    this.pendingPromise = makeAbortablePromise(fetchActivities(nextPage));
     this.pendingPromise.promise
       .then(() => this.setState({ pending: false }))
       .catch(err => {
         if (!(err instanceof AbortPromiseException)) {
           this.setState({ pending: false });
         }
       });
-  }
-
-  componentWillUnmount() {
-    if (this.pendingPromise) {
-      this.pendingPromise.abort();
-    }
-  }
+  };
 
   render() {
-    const { activities } = this.props;
+    const {
+      activities,
+      intl: { formatMessage },
+      isNextPage
+    } = this.props;
     const { pending } = this.state;
 
     return (
       <div className="activities-list">
-        <h2 className="activities-list__heading">Activities</h2>
+        <h2 className="activities-list__heading">
+          {formatMessage({
+            id: 'activity.title'
```

Comment:
I thought about it, but activity id for adding an item is already implemented on the dev server and doesn't have this prefix. So I decided to avoid DB's migration.